oidc-client/src/main/java/org/glite/authz/oidc/client/service/HttpService.java

```diff
@@ -0,0 +1,13 @@
+package org.glite.authz.oidc.client.service;
+
+import org.springframework.util.MultiValueMap;
+
+public interface HttpService {
+
+  public String postWithBasicAuthentication(String endpoint, String username, String password,
```

Comment:
This could be at an higher level of abstraction, and have signature:
```java
Optional<String> introspectToken(String accessToken);
Optional<String> getUserInfoForToken(String accessToken);
```

Reply:
and add Javadoc